pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -232,6 +250,99 @@ func getMachineDefaulterOperation(platformStatus *osconfigv1.PlatformStatus) mac
 	}
 }
 
+// NewValidatingWebhookConfiguration creates empty validation webhook configuration
+func NewValidatingWebhookConfiguration() *admissionregistrationv1.ValidatingWebhookConfiguration {
+	return &admissionregistrationv1.ValidatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: webhookConfigurationName,
+			Annotations: map[string]string{
+				"service.beta.openshift.io/inject-cabundle": "true",
+			},
+		},
+		Webhooks: []admissionregistrationv1.ValidatingWebhook{
+			MachineValidatingWebhook(defaultWebhookServiceNamespace),
+			MachineSetValidatingWebhook(defaultWebhookServiceNamespace),
+		},
+	}
+}
+
+// MachineValidatingWebhook returns validating webhooks for machine to populate configuration
+func MachineValidatingWebhook(namespace string) admissionregistrationv1.ValidatingWebhook {
+	machineServiceReference := admissionregistrationv1.ServiceReference{
+		Namespace: namespace,
+		Name:      webhookName,
+		Path:      pointer.StringPtr(DefaultMachineValidatingHookPath),
+	}
+	return admissionregistrationv1.ValidatingWebhook{
+		AdmissionReviewVersions: []string{"v1beta1"},
+		Name:                    "validation.machine.machine.openshift.io",
+		FailurePolicy:           &webhookFailurePolicy,
+		SideEffects:             &webhookSideEffects,
+		ClientConfig: admissionregistrationv1.WebhookClientConfig{
+			Service: &machineServiceReference,
+		},
+		Rules: []admissionregistrationv1.RuleWithOperations{
+			{
+				Rule: admissionregistrationv1.Rule{
+					APIGroups:   []string{machine.GroupName},
+					APIVersions: []string{SchemeGroupVersion.Version},
+					Resources:   []string{"machines"},
+				},
+				Operations: []admissionregistrationv1.OperationType{
+					admissionregistrationv1.Create,
+					admissionregistrationv1.Update,
+				},
+			},
+		},
+	}
+}
+
+// NewMutatingWebhookConfiguration creates empty mutating webhook configuration
+func NewMutatingWebhookConfiguration() *admissionregistrationv1.MutatingWebhookConfiguration {
+	return &admissionregistrationv1.MutatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: webhookConfigurationName,
+			Annotations: map[string]string{
+				"service.beta.openshift.io/inject-cabundle": "true",
+			},
+		},
+		Webhooks: []admissionregistrationv1.MutatingWebhook{
+			MachineMutatingWebhook(defaultWebhookServiceNamespace),
+			MachineSetMutatingWebhook(defaultWebhookServiceNamespace),
+		},
+	}
+}
+
+// MachineMutatingWebhook returns mutating webhooks for machine to apply in configuration
+func MachineMutatingWebhook(namespace string) admissionregistrationv1.MutatingWebhook {
```

Comment:
why is the namespace a parameter here?

Reply:
Technically it should be passed all along from config, but let's use the default.